app/lib/three_scale/spam_protection/integration/controller.rb

```diff
@@ -1,27 +1,74 @@
+# frozen_string_literal: true
+
 module ThreeScale::SpamProtection
+
   module Integration
 
     module Controller
+      extend ActiveSupport::Concern
+
+      class_methods do
+
+        def enabled_checks
+          _enabled_checks
+        end
+
+        def has_spam_protection(*checks)
+          self._enabled_checks = checks.presence || AVAILABLE_CHECKS
+        end
+      end
+
+      included do
+        delegate :spam?, :spam_probability, to: :spam_protection
+        class_attribute :_enabled_checks, instance_reader: false, instance_writer: false
+        before_action :instantiate_checks
+      end
+
+      def instantiate_checks
+        spam_protection_conf.enable_checks! self.class.enabled_checks
+      end
+
+      def spam_protection
+        @spam_protection ||= ThreeScale::SpamProtection::Protector.new(self)
+      end
+
+      def spam_protection_conf
+        @spam_protection_conf ||= ThreeScale::SpamProtection::Configuration.new(self)
+      end
+
+      def spam_protection_form(form)
+        ThreeScale::SpamProtection::FormProtector.new(form, self)
+      end
+
+      def store
+        @store ||= SessionStore.new(request.session)
+      end
+
+      def level
+        site_account.settings.spam_protection_level
+      end
+
       private
 
-      def spam_check_save(object, &block)
-        block_given? ? yield : object.save if spam_check(object)
+      def spam_check_save(object)
+        (block_given? ? yield : object.save) if spam_check(object)
       end
 
       def verify_captcha(object)
         verify_recaptcha(model: object, attribute: :recaptcha)
```

Comment:
Not relevant, as this part has not changed, but from what I saw, it doesn't really make much sense to pass the model and the attribute to `verify_recaptcha`, as even though it adds an error to the model passed to it, we don't actually show this error anyway :upside_down_face: 

in fact, it could make more sense to call this without arguments, so a flash message would be shown.
The downside of it is that on the password reminder form there is another flash message, so there would be redundant flashes (one with proper "error" red style, and another "normal").

But anyway, this does not need to be touched in this PR, it's just something I noticed.

Reply:
You're right, it's better to remove that `object` param if it's being ignored anyway. So we simplify the logic which is complicated enough already. I tried locally and it causes a little mess on the UI which needs time to work on. So better leave it for another PR

Reply:
Yeah, I think there are several things that could be improved, but indeed, it's better we don't change the actual behavior here.